docs/authorization/authority.md

```diff
@@ -0,0 +1,127 @@
+# Authority management
+Netgrif Application Engine implements authority objects to manage access
+to resources, to protect resources from unauthorized access. Then these 
+authorities can be assigned to users to provide access to those secured
+resources.
+
+## Authorizing objects
+Authorizing objects are Java enum values, that represent authorities,
+that are needed to access a resource or use an action. E.g. to be able to import new 
+process (Petri Net) into the application, the user must have authority 
+created from `AuthorizingObject.PROCESS_UPLOAD` authorizing object.
+
+These authorizing objects are predefined, and they are used to create the 
+`Authority` objects/entities at application startup. There followings are predefined:
+
+- `PROCESS_UPLOAD`- to import new process
+- `PROCESS_VIEW_ALL`- to retrieve all the processes imported by any user
+- `PROCESS_VIEW_MY`- to retrieve only processes imported by logged user
+- `PROCESS_DELETE_ALL`- to delete processes imported by any user
+- `PROCESS_DELETE_MY`- to delete processes imported by logged user
+- `FILTER_UPLOAD`- to upload filter
+- `FILTER_DELETE_MY`- to delete filter imported by logged user
+- `FILTER_DELETE_ALL`- to delete filter imported by any user
+- `USER_CREATE`- to invite or create user
+- `USER_DELETE`- to remove user
```

Comment:
I have split the authority to the required two authorities. In the example app and in the base engine there is no endpoint for deleting users, these authorities are only used over actions in action delegate.

Do we want to enable to delete own accounts on actions? Is there a chance that it will cause a problem, when you are calling delete action for own account over assigned task (the task can be assigned to own user)?